### ISSUE TITLE
JSON objects are sent using single quotes causing JSON parsing failure on browser side

### ISSUE THREAD

**Issue by deepio-oc** [sysid/sse-starlette#21]
I am trying to send a dictionary object to the browser but JSON.parse fails on browser side saying single quotes are not allowed.

The issue is with the following code, which tries to convert dictionary to string (resulting in single quotes). 
```
        for chunk in self.LINE_SEP_EXPR.split(str(self.data)):
            buffer.write(f"data: {chunk}")
            buffer.write(self._sep)
```

Could you check the data type and may use json.dumps or provide it as configuration option?

**Comment by pavlo-yashchuk** [comment]
Yes, deepio-oc right
When I transfer json data, in ServerSentEvent.encode the dictionary turns into a string, but I want to transfer json file :(

**Comment by deepio-oc** [comment]
@pavlo-yashchuk dictionary need to be encoded as string before it is sent. On the receiver side, you need to decode it back to json object from string.

my issue is that encoded string format is not JSON compliant and I cant decode it back.

@sysid I dont see any option to open this ticket.

**Comment by pavlo-yashchuk** [comment]
Okay) but  I found solution for me
add json loads before dumps
![image](https://user-images.githubusercontent.com/85494277/131511211-b825cd7b-af43-410a-86cb-ea51f420791c.png)

**Comment by CiberNin** [comment]
I don't think this should be marked closed. Converting dicts to valid JSON seems like a reasonable task for the library to handle, and not a particularly hard feature to add.
For example, right now simple code like this won't work. 
![image](https://user-images.githubusercontent.com/59232386/185147434-a0947d5b-cc0c-4c9d-b7ce-ffa0f47e51e6.png)

**Comment by wiseyoungbuck** [comment]
This seems like it should be fixed.
